src/core/components/object.js

```diff
@@ -242,19 +242,21 @@ module.exports = function object (self) {
       }
 
       function next () {
-        self._ipld.put(node, multicodec.DAG_PB, {
-          cidVersion: 0,
-          hashAlg: multicodec.SHA2_256
-        }).then(
-          (cid) => {
-            if (options.preload !== false) {
-              self._preload(cid)
-            }
-
-            callback(null, cid)
-          },
-          (error) => callback(error)
-        )
+        self._gcLock.readLock((cb) => {
```

Comment:
I might have missed the explanation, can someone link me to it or explain here why does GC need another lock on top of the Repo lock?

Reply:
The GC lock is used to prevent blocks being added/removed from the blockstore while GC is running. My understanding is that the Repo lock is used to prevent multiple instances of js-ipfs-repo from opening the same path, is that correct?

Reply:
So what happens to all inflight blocks coming in Bitswap? Is there a test to see if js-ipfs OOMs when performing a GC and someone sends a ton of blocks through bitswap that get queued in memory?

Reply:
That's a good question - in general that's the problem with stop-the-world GC, it blocks everything up while GC is being performed.